polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -228,7 +248,10 @@ $loading-panel-height: 53px;
       &,
       .TableCell-first,
       .TableCell-first + .TableCell {
-        background-color: var(--p-color-bg-subdued-hover);
+        // deliberately moved it down in the cascade because otherwise the
+        // selected state overrides the hover state and I think we want it the
+        // other way around.
+        background-color: var(--p-color-bg-hover);
```

Comment:
I think we'll need to account for the Zebra strip variation here and on line `269` with the `var(--p-color-bg-primary-subdued-hover)` value.

The background seems different from the rest of the cells on hover when selected and when not selected.

https://github.com/Shopify/polaris/assets/11774595/e1fffd4e-fc8a-4095-9673-83b9b511167f

Reply:
We've punted design updates for Zebra striping: It's not in use in web and there's a desire to deprecate it in the future, so there's no new designs for what colours should be used on zebra when combined with status=subdued/selected/etc.

For illustrative purposes, I've gathered up all the possible states we'd need to cover in a table:

<details>

<summary>Logic matrix</summary>

|                    | [base]                                                                                                                                                                                                                                                                                                                                                                                                                                          | `subdued`                                                                                                                                                                                                                                                                                                                                                                                                                                       | `zebra`                                                                                                                                                                                                                                                                                                                                                                                                                                          | `status="success"`                                                                                                                                                                                                                                                                                                                                                                                                                              | `status="subdued"`                                                                                                                                                                                                                                                                                                                                                                                                                              |
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [base]             | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `subdued`          | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `zebra`            | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `status="success"` | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table>  | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |
| `status="subdued"` | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table>  <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> | <table> <thead> <tr> <th></th> <th>[base]</th> <th>hovered</th> <th>selected</th> <th>hovered + selected</th> </tr> </thead> <tbody><tr> <td>[base]</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>selected</td> <td></td> <td></td> <td></td> <td></td> </tr> <tr> <td>hovered + selected</td> <td></td> <td></td> <td></td> <td></td> </tr> </tbody></table> |


</details>

You can see by removing Zebra striping, we are able to leave off ~80 variants 🎉

Reply:
Awesome! That sounds great, I like the simplification 👍 

Thanks for the context @jesstelford and @sarahill